WPI Result page

CharlottePerin opened the pull request:
Ajout du template pour la page de résultats détaillés et les fonctions associées dans views.py


- [ ] virer printNumbers&Sequence.py

CharlottePerin committed:
Create printNumbers&Sequence.py

Juste pour avoir la bout de code qui permet d'aligner comme il faut les chiffres au-dessus d'une séquence.
ABSOLUMENT PAS automatisé ni rien (le fait pour un cas d'exemple), il s'agit juste d'avoir une trace,
que tout le monde y ait accès

CharlottePerin committed:
Add files via upload

CharlottePerin committed:
Update views.py

Add the view for the detailed results page and the function that creates the string containing the aligned positions in sequence

CharlottePerin committed:
Merge branch 'develop' into result_page

CharlottePerin commented:
Il faudrait virer printNumbers&Sequence.py mais je vois pas comment faire alors bon, c'est dommage

nanls committed:
Trailing spaces

nanls committed:
Fix argument retriving from url

nanls committed:
Sqlalchemize a querry

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+    ARGUMENT:
+        length of the sequence to consider.
+    """
+    numbers = range(0, 10000, 10)
```
Oulà, ca sent la fatigue : ya 2-3 trucs qui collent pas. XD 

- en Python 3, ya pas mal de truc sur lequel ca clash

- pourquoi générer une liste grande, alors qu etu connais la longueur ?

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+        length of the sequence to consider.
+    """
+    numbers = range(0, 10000, 10)
+    numbers[0] = 1
```
en python 3, un range ne donne pas une liste mais un range, qui est immuable, donc tu peux pas changer la valeur apres coup. 

Donc à la rigueur, je te propose un : 
` numbers = [1].extend(  range(10, length, 10) ) `
et du coup ca fait une liste et pas un range.

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+    pos = ""
+    for i in xrange((length/10)+1):
+        if i == 0:
+            pos += "{:<9d}".format(numbers[0])
```
faire un if alors que tu sais qu'il va s'effectuer que la premiere fois, c'est dommage, je te propose un 

`pos = "{:<9d}".format(numbers[0])` (ou meme directement : `pos = 1`) 
puis de commencer la boucle en concaténant la suite.

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+    numbers = range(0, 10000, 10)
+    numbers[0] = 1
+    pos = ""
+    for i in xrange((length/10)+1):
```
en Python 3 : les xrange ,n'existent pas, les ranges font le boulot.

nanls replied:
et aussi, en utilisant lenght pour générer numbers, ya pas besoin de ce range là, du coup. \o/ :D

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+        if i == 0:
+            pos += "{:<9d}".format(numbers[0])
+        else:
+            pos += "{:<10d}".format(numbers[i])
```
en Python 3, je crois qu'on peut pas acceder à un element comment ca. 
Mais du coup, en utilisant number dans le for, ya pas besoin de l'indexation.

nanls replied:
En fait, en ne mettant pas de 1 et en alignat à droite plutot qu'à gauche, on peut même supprimer qq lignes de code. (genre tout ce qui concerne le premier élément).

nanls reviewed (changes requested):
Je me rends compte que 'st tout fouilli, alors en bref, voilà la fonction qui prend en compte toutes mes remarques : 


```
def positionsPrinter(length):
    """
    Create a string containing the position numbers in a way that they will be
    aligned with the sequence.
    ARGUMENT:
        length of the sequence to consider.
    """
    numbers = range(10, length + 1, 10)
    pos =""
    for number in numbers:
        pos += "{:>10d}".format(number )
    return pos + "\n"

```

nanls committed:
Python3nization of positionsPrinter

nanls committed:
SQLAlchemize a query

nanls committed:
Adapt the output from db to the argument ramachandran.compute_ramachandran_map is waiting for

from [ (phi, psi) for each angle ]
to [ (phi for each angle), (psi for each angle) ]

nanls committed:
Ramap computation function will return the path of the ramap

nanls commented on app/views.py:
```diff
@@ -41,6 +41,48 @@ def upload():
+    annotations = model.Annotation.select(model.Annotation.pdb_id= \
+        PDBid).execute().all()
+    for meth in annotations:
+        annot["{<:7s}".format(annotations["method"])] = annotations["result"]
```
je suis pas sure de comprendre ce que tu voulais faire. Mais là je sers plus à rien, alors je pose ça là, et go dodo ! \o/ :D 

@CharlottePerin  : a la fin si t'as un tableaux d’objet `Annotation`, pour lequels on pourra accéder aux attributs dans le template, ça irait, non ?

CharlottePerin replied:
ah! alors:
je récupère toutes les annotations disponibles pour un pdb
puis
je crée un dictionnaire construit tel que, par example:
{"DSSP   ": "rkeqbgkerglergkrkkkkr"}
donc nom de la méthode sur 7 caractères (pour l'affichage): annotation
ce dictionnaire sera ensuite parcouru et affiché dans le template associé

CharlottePerin replied:
et il a pas aimé mais j'ai vraiment écrit: "DSSP" avec 3 espaces ensuite

nanls replied:
okokok

CharlottePerin commented:
J'ai pas vraiment compris tes remarques là
enfin en gros, il y avait un comm au dessus de if esle avant qui disait "je sais pas pourquoi, mais c'est comme ça qu'il veut sinon, ça ne marche pas"
et concernant ce code, c'était bien "{:<}" et non pas "{:>}"

Athenais commented:
haha, j'avoue que c'était plus très clair. XD

CharlottePerin commented:
Du coup, le code est bon non?

Athenais commented:
Ben celui qu'il y avait marchait pas (pour toute les raisons liées à Python 2 / 3). Mais maintenant, oui. Hésitez pas à tester, d'ailleurs.

CharlottePerin committed:
Update views.py

Modifs de positions Printer pour que ce soit compatible avec python 3

CharlottePerin commented:
Le code marche complètement mainteannt (tu commençais les positions à 10 plutôt que 1)

Athenais commented:
ouais, je m’étais qu'on s'en foutait du 1. Mais merci d'avoir pris le temps de corriger. :)

nanls commented on app/views.py:
```diff
@@ -41,6 +41,45 @@ def upload():
+        length of the sequence to consider.
+    """
+    numbers = range(10, length + 1, 10)
+    pos = "{:<9d}".format(1)
```
Merci, @CharlottePerin . :)

Athenais commented:
bouhouhou, c'était pas fini, @jecarvaill :'(

jecarvaill commented:
Je ne savais pas, j'ai vu qu'il fallait merge. Je pensais bien faire 😫

2017-04-12 15:32 GMT+02:00 Athénaïs Vaginay <notifications@github.com>:

> bouhouhou, c'était pas fini, @jecarvaill <https://github.com/jecarvaill>
> :'(
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Athenais/M2BI-db-project/pull/52#issuecomment-293577079>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AYsi5F56PqT4M9-uMPqhq3o8XhB4H_nIks5rvNJ8gaJpZM4M6oFa>
> .
>
